analysis/crystal/crystal.go

```diff
@@ -0,0 +1,85 @@
+package crystal
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+// Host is the base URL for the analysseur API.
+var Host string
+
+type request struct {
+	ID       string `json:"id"`
+	Contents string `json:"contents"`
+}
+
+type response struct {
+	ID       string    `json:"id"`
+	Problems []problem `json:"problems"`
+	Error    string    `json:"error"`
+}
+
+type problem struct {
+	Type   string `json:"type"`
+	Result string `json:"result"`
```

Comment:
I notice in your test that the result is a boolean. Will this sometimes be other things?